submissions/Ant-C-tech/tiny-js-world/index.js

```diff
@@ -0,0 +1,135 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/Ant-C-tech/a-tiny-JS-world
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+const dog = {
+   species: 'dog',
+   name: 'Spike',
+   gender: 'male',
+   legs: 4,
+   hands: 0,
+   saying: 'Woof-Woof!',
+   friends: ['John']
+}
+
+const cat = {
+   species: 'cat',
+   name: 'Tom',
+   gender: 'male',
+   legs: 4,
+   hands: 0,
+   saying: 'Meow-Meow!',
+   friends: ['John']
+}
+
+const woman = {
+   species: 'human',
+   name: 'Jane',
+   gender: 'female',
+   legs: 2,
+   hands: 2,
+   saying: 'Give me your salary!'
+}
+
+const man = {
+   species: 'human',
+   name: 'John',
+   gender: 'male',
+   legs: 2,
+   hands: 2,
+   saying: 'Give me your clothes, your boots and your motorcycle!',
+   friends: ['Spike', 'Tom']
+}
+
+function createGreetingContent(obj) {
+
+   let content = ''
+   let interlocutor = ''
+   
+   switch (true) {
+      case obj.species == 'dog':
+         interlocutor = 'a cat'
+         break;
+      case obj.species == 'cat':
+         interlocutor = 'a dog'
+         break;
+      case obj.species == 'human' && obj.gender == 'male':
+         interlocutor = 'a biker'
+         break;
+      case obj.species == 'human' && obj.gender == 'female':
+         interlocutor = 'my husband'
+         break;
+   }
+
+   Object.keys(obj).forEach((key) => {
+      switch (true) {
+         case key == 'species':
+            content += `Hello! I am a ${obj[key]}. `
+            break;
+         case key == 'name':
+            content += `My name is ${obj[key]}. `
+            break;
+         case key == 'gender':
+            content += `My gender is ${obj[key]}. `
+            break;
+         case key == 'legs':
+            if (obj[key] > 0) {
+               content += `I have ${obj[key]} legs. `
+            } else {
+               content += `I have no legs. `
+            }
+           
+            break;
+         case key == 'hands':
+            if (obj[key] > 0) {
+               content += `I have ${obj[key]} hands. `
+            } else {
+               content += `I have no hands. `
+            }
+            break;
+         case key == 'saying':
+            content += `If I see ${interlocutor}, I say "${obj[key]}". `
+            break;
+         case key == 'friends':
+            if (obj[key].length > 1) {
+               content += `I have friends: ${obj[key]}.`
+            } else {
+               content += `I have friend: ${obj[key]}.`
+            }
+
+            break;
+         default:
+            break;
+      }
```

Comment:
Using conditions in `switch` `case` statemnets is antipattern and should be avoided by all means.
Consider the following:
```
const propPrefixes = {
  hands: 'I have % hands',
  legs: 'I have % legs',
};

const content = ['legs', 'hands']
  .map(propName => prefixes[propname].replace('%', obj[propName]))
  .join('. ');
```
You'd need to devise an approach to handle friends, but let be your challenge to take.

```